### PR TITLE
[Frontend][MXNet] argmax, argmin ops support

### PR DESCRIPTION
**Pull request by imorinaga**
This PR supports argmax and argmin for MXNet. Since output dtype is fp32, I added dtype interface to nnvm.

@tqchen @nishi-t @masahi could you review? Thanks!

### CODE REVIEW

File: nnvm/src/top/tensor/reduce.cc [hunk 1]
```diff
@@ -302,6 +300,10 @@ values over a given axis.
     TShape r_axes = GetReduceAxes(inputs[0]->shape.size(),
                                   param.axis, param.exclude);
     auto axis = ShapeToArray(r_axes);
+    if (param.dtype == kFloat32)
+      return Array<Tensor>{
+        topi::cast(topi::argmax(
+        inputs[0], axis, param.keepdims), out_info[0]->dtype)};
     return Array<Tensor>{
       topi::argmax(inputs[0], axis, param.keepdims) };
```
Comment (masahi) [inline nnvm/src/top/tensor/reduce.cc]: better to factor two identical topi::argmax(...) into one

File: nnvm/src/top/tensor/reduce.cc [hunk 2]
```diff
@@ -324,6 +326,10 @@ values over a given axis.
     TShape r_axes = GetReduceAxes(inputs[0]->shape.size(),
                                   param.axis, param.exclude);
     auto axis = ShapeToArray(r_axes);
+    if (param.dtype == kFloat32)
+      return Array<Tensor>{
+        topi::cast(topi::argmin(
+        inputs[0], axis, param.keepdims), out_info[0]->dtype)};
     return Array<Tensor>{
       topi::argmin(inputs[0], axis, param.keepdims) };
```
Comment (masahi) [inline nnvm/src/top/tensor/reduce.cc]: the same comment as argmax

File: topi/python/topi/cuda/reduction.py [hunk 3]
```diff
@@ -107,7 +107,10 @@ def traverse_before_reduce(operator):
     def traverse_after_reduce(operator):
         """Internal travserse function"""
         if tag.is_broadcast(operator.tag):
-            raise RuntimeError("Not yet support ewise after reduce")
+            for tensor in operator.input_tensors:
+                if tensor.op not in scheduled_ops:
+                    traverse_after_reduce(tensor.op)
+            _schedule_injective(operator, sch)
```
Comment (masahi) [inline topi/python/topi/cuda/reduction.py]: sorry I don't understand this change. What is it for?

File: nnvm/tests/python/frontend/mxnet/test_forward.py [hunk 4]
```diff
@@ -174,6 +174,15 @@ def test_forward_zeros_like():
     data = mx.sym.var('data')
     mx_sym = mx.sym.zeros_like(data, dtype='float32')
     verify_mxnet_frontend_impl(mx_sym, (2, 3, 4), (2, 3, 4))
+
+def test_forward_argmax():
+    data = mx.sym.var('data')
+    mx_sym = mx.sym.argmax(data, axis=1)
+    verify_mxnet_frontend_impl(mx_sym, (5, 3), (5,))
+def test_forward_argmin():
```
Comment (masahi) [inline nnvm/tests/python/frontend/mxnet/test_forward.py]: add a black line

File: topi/python/topi/cuda/reduction.py (hunk 3 above)
Comment (imorinaga) [reply topi/python/topi/cuda/reduction.py]: This code supports fused operators which perform an elemwise op after a reduce operator for cuda. 

For another example, nnvm.symbol.mean cannot compilable for cuda. 
```
import nnvm
import nnvm.symbol as sym

x = sym.Variable("x")
y = sym.mean(x, axis=1)
compute_graph = nnvm.graph.create(y)
shape = (4,3, )
deploy_graph, lib, params = nnvm.compiler.build(
    compute_graph, target="cuda", shape={"x": shape}, dtype="float32")
```
The Error is as following. 

  File "/home/morinaga/.local/lib/python2.7/site-packages/topi-0.5.dev0-py2.7.egg/topi/cuda/reduction.py", line 114, in traverse_after_reduce
    raise RuntimeError("Not yet support ewise after reduce")
RuntimeError: Not yet support ewise after reduce

**Comment by tqchen** [comment]
Thanks @imorinaga @masahi this is merged

**Pull request merged**